hive/src/hex.rs

```diff
@@ -2,9 +2,9 @@ use std::collections::HashSet;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct Hex {
-    pub x: i64,
-    pub y: i64,
-    pub z: i64,
+    pub x: i8,
```

Comment:
i haven't done the math on it, but do you think `i8`s limit the possible boardspace significantly? i suspect you'd have to be playing pathologically to get outside `i8` bounds, but still

Reply:
There are only 28 tiles in the game, and in a normal game the entire hive doesn't migrate. It is possible to have two queens and two beetles slowly migrate an indefinite distance by crawling past each other, but this doesn't seem like an important usecase.